src/CameraServer.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <cstring>
+#include <set>
+#include <sstream>
+#include <string.h>
+
+#include "CameraComponent_V4L2.h"
+#include "CameraServer.h"
+#include "log.h"
+#include "mavlink_server.h"
+#include "util.h"
+
+#define DEVICE_PATH "/dev/"
+#define VIDEO_PREFIX "video"
+#define DEFAULT_SERVICE_PORT 8554
+
+CameraServer::CameraServer(ConfFile &conf)
+    : mavlink_server(conf, streams, rtsp_server)
+    , rtsp_server(streams, DEFAULT_SERVICE_PORT)
+    , cameraCount(0)
+{
+    cameraCount = detectCamera(conf);
```

Comment:
The functionality will not be limited to V4L2 cameras. Support will be added for real sense cameras too. However to do that, I first need to understand the controls and parameters of RS camera that can be exposed to user. Also, understand the APIs to  be able to set the parameters. Finally, make the changes. As a first step, decided to make the feature work for V4L2 and then take up RS.